pgjdbc/src/main/java/org/postgresql/ssl/jdbc4/LibPQFactory.java

```diff
@@ -252,6 +258,29 @@ public boolean verify(String hostname, SSLSession session) {
     }
     // Extract the common name
     X509Certificate serverCert = peerCerts[0];
+
+    try {
+      // Check for Subject Alternative Names (see RFC 6125)
+      Collection subjectAltNames = serverCert.getSubjectAlternativeNames();
+
+      if (subjectAltNames != null) {
+
+        for (Iterator<List> sanit = subjectAltNames.iterator(); sanit.hasNext();) {
+
+          List list = sanit.next();
+          Integer type = (Integer) list.get(0);
+          String san = (String) list.get(1);
+
+          // this mimics libpq check for ALT_DNS_NAME
+          if (type != null && san != null && type == ALT_DNS_NAME && san.equals(hostname)) {
```

Comment:
Please add `.toLowerCase` hostname normalization.
`hostname.toLowerCase` should be outside of a loop )

Reply:
IMO, an `.equalsIgnoreCase(hostname)` should be safer, or a `.toLowerCase(Locale.ENGLISH)` to san.

see http://www.i18nguy.com/unicode/turkish-i18n.html

Reply:
Thanks, makes sense.  👍  for `equalsIgnoreCase`